main.rb

```diff
@@ -0,0 +1,157 @@
+require './book'
+require './classroom'
+require './corrector_class'
+require './person_class'
+require './rental'
+require './teacher_class'
+require './student_class'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+    @options = {
+      '1' => 'List all books',
+      '2' => 'List all people',
+      '3' => 'Create a person',
+      '4' => 'Create a book',
+      '5' => 'Create a rental',
+      '6' => 'List all rentals for a given person',
+      '7' => 'Exit'
+    }
+  end
+
+  def run
+    puts '>>>>>>>>>>>>> Welcome to OOP School Library <<<<<<<<<<<<<'
+
+    loop do
+      puts ''
+      puts 'Please select an option by entering corresponding number to the given list of values'
+      @options.each { |key, value| puts "#{key} - #{value}" }
+      option = gets.chomp
+      break if option.eql?('7')
+
+      puts "\nYou selected: #{option}\n"
+
+      menu(option)
+    end
+  end
+
+  def menu(option)
+    case option
+    when '1'
+      books
+    when '2'
+      people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      rentals
+    else
+      puts 'Please select a valid option'
+    end
+  end
+
+  private
+
+  def books
+    if @books.empty?
+      puts "\nLibrary is empty"
+    else
+      @books.each { |book| puts book }
+    end
+  end
+
+  def people
+    if @people.empty?
+      puts "\nNobody registered here ..."
+    else
+      @people.each { |person| puts person }
+    end
+  end
+
+  def create_person
+    puts "\nDo you want to create a student (1) or a teacher (2)? [Input the number]: "
+    option = gets.chomp
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Please enter a valid option'
+    end
+    person = option == '1' ? 'Student' : 'Teacher'
+    puts "#{person} successfully created!"
```

Comment:
- [ ] These lines will always run when you call `create_person `method and this return bad behaviour when the option passed is different from `1` or `2`. For example, if I pass `3` as the option it will print `Teacher successfully created!` although it isn't the case. Let's fix that

![Screenshot from 2022-01-08 20-04-55](https://user-images.githubusercontent.com/40503704/148657036-acf825aa-0aea-42fb-83cb-a4e4e89f4ce8.png)

Reply:
Thanks for pointing out my mistake and I've resolved this issue.